opm/simulators/flow/TracerModel.hpp

```diff
@@ -499,7 +500,7 @@ class TracerModel : public GenericTracerModel<GetPropType<TypeTag, Properties::G
         }
 
         const Scalar dt = simulator_.timeStepSize();
-        const std::size_t well_index = eclWell.seqIndex();
+        const std::size_t well_index = simulator_.problem().wellModel().wellState().index(well.name()).value();
         const auto& ws = simulator_.problem().wellModel().wellState().well(well_index);
```

Comment:
The `well_index` isn't really used anywhere else, so you might as well use the [name based lookup](https://github.com/OPM/opm-simulators/blob/c7c0aeb4e10509d3dce8357f32fdcdc9c5f405c1/opm/simulators/wells/WellState.hpp#L332) for `ws`:
```C++
const auto& ws = simulator_.problem().wellModel().wellState.well(well.name());
```
That will bypass this problem completely.

Reply:
Indeed, if that is available.. This is basically reverting to the old code, I mixed up the well state index and the seq index (which will match in serial only). Have applied this improvement.

Reply:
The new structure is a change from ```map<pair<well_name, tracer_name>, Scalar>``` to ```map<well_index, struct_with_tracer_name_and_scalar>```

Reply:
> The new structure is a change from `map<pair<well_name, tracer_name>, Scalar>` to `map<well_index, struct_with_tracer_name_and_scalar>`

Sure, I appreciate that and it's a clear improvement, but the `WellState` uses numbering relative to `wells_ecl_` which is local to each process and there's no direct way to compute that index if you "just" have a `Well` object.  If on the other hand you have a `WellInterfacePtr`, then you can use `indexOfWell()` as the lookup index.

Reply:
yes, as I now learned the hard way.

Reply:
> yes, as I now learned the hard way.

Right, and my point is that if you're not going to use that index for anything else, you might as well just use the name-based lookup.